### PR TITLE
Adding issue state proto

### PR DESCRIPTION
**Pull request by michelle192837**
TestGrid uses this to add linked issues to tests.

### CODE REVIEW

File: testgrid/issue_state.proto [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+syntax = "proto3";
+
+message IssueInfo {
+  string issue_id = 1;
+  string title = 2;  // Issue title or description.
+
+  reserved 3, 4;
+
+  repeated string targets = 5;  // Associated test targets (mentioned in the issue).
```
[the remark this answers is not in the document]
Comment (michelle192837) [reply testgrid/issue_state.proto]: 'results' isn't quite right here; this should be the test name (which is why I used target here) so we know what to link it to, whereas 'result' is more the status. ('id' should be the corresponding thing here probably, but just using 'id' seems too general. Maybe 'test_id'?)

[the remark this answers is not in the document]
Comment (fejta) [reply testgrid/issue_state.proto]: We use row name in the state proto, and these should match: https://github.com/kubernetes/test-infra/blob/5c5fc3f59bbd46a3971d1e9c06f024670f375146/testgrid/state.proto#L22-L23

[the remark this answers is not in the document]
Comment (michelle192837) [reply testgrid/issue_state.proto]: Updated; updater uses the equivalent of Row.id to associate bugs, so tracking that for now.

**Pull request merged**